src/core/public/chrome/ui/header/header_help_menu.tsx

```diff
@@ -227,127 +263,90 @@ class HeaderHelpMenuUI extends Component<Props, State> {
+
+    const getFeedbackText = () =>
+      i18n.translate('core.ui.chrome.headerGlobalNav.helpMenuGiveFeedbackOnApp', {
+        defaultMessage: 'Give feedback on {appName}',
+        values: { appName: helpExtension.appName },
+      });
+
+    const customLinks =
+      links &&
+      links.map((link, index) => {
+        const addSpacer = index < links.length - 1;
+        switch (link.linkType) {
+          case 'documentation': {
+            const { linkType, ...rest } = link;
+            return createCustomLink(
+              index,
+              <FormattedMessage
+                id="core.ui.chrome.headerGlobalNav.helpMenuDocumentation"
+                defaultMessage="Documentation"
+              />,
+              addSpacer,
+              {
                 target: '_blank',
                 rel: 'noopener',
                 ...rest,
-              });
-            case 'custom':
-              return this.createCustomLink(index, text, index < links.length - 1, { ...rest });
-            default:
-              break;
+              }
+            );
           }
-        });
-
-      customContent = (
-        <>
-          <EuiTitle size="xxs">
-            <h3>{appName}</h3>
-          </EuiTitle>
-          <EuiSpacer size="s" />
-          {customLinks}
-          {content && (
-            <>
-              {customLinks && <EuiSpacer size="s" />}
-              <HeaderExtension extension={content} />
-            </>
-          )}
-        </>
-      );
-    }
-
-    const button = (
-      <EuiHeaderSectionItemButton
-        aria-expanded={this.state.isOpen}
-        aria-haspopup="true"
-        aria-label={intl.formatMessage({
-          id: 'core.ui.chrome.headerGlobalNav.helpMenuButtonAriaLabel',
-          defaultMessage: 'Help menu',
-        })}
-        onClick={this.onMenuButtonClick}
-      >
-        <EuiIcon type="help" size="m" />
-      </EuiHeaderSectionItemButton>
-    );
+          case 'github': {
+            const { linkType, labels, title, ...rest } = link;
+            return createCustomLink(index, getFeedbackText(), addSpacer, {
+              iconType: 'logoGithub',
+              href: createGithubUrl(labels, title),
+              target: '_blank',
+              rel: 'noopener',
+              ...rest,
+            });
+          }
+          case 'discuss': {
+            const { linkType, ...rest } = link;
+            return createCustomLink(index, getFeedbackText(), addSpacer, {
+              iconType: 'editorComment',
+              target: '_blank',
+              rel: 'noopener',
+              ...rest,
+            });
+          }
+          case 'custom': {
+            const { linkType, content: text, href, ...rest } = link;
+            return createCustomLink(index, text, addSpacer, {
+              href,
+              onClick: this.createOnClickHandler(href, navigateToUrl),
+              ...rest,
+            });
+          }
```

Comment:
Only functional change of the PR: we now add a `onClick` handler to custom links to perform SPA navigation when the link is internal.